trin-core/src/portalnet/types/content_key.rs

```diff
@@ -96,14 +100,67 @@ pub struct BlockBody {
 }
 
 /// A key for the transaction receipts for a block.
-#[derive(Clone, Debug, Decode, Encode)]
+#[derive(Clone, Debug, Decode, Encode, PartialEq)]
 pub struct BlockReceipts {
     /// Chain identifier.
     pub chain_id: u16,
     /// Hash of the block.
     pub block_hash: [u8; 32],
 }
 
+/// A key for the master header accumulator.
+#[derive(Clone, Debug, Decode, Encode, PartialEq)]
+#[ssz(enum_behaviour = "union")]
+pub enum MasterAccumulator {
+    Latest(SszNone),
+    MasterHash(H256),
+}
+
+/// Struct to represent encodable/decodable None value for an SSZ enum
+#[derive(Clone, Debug, PartialEq)]
+pub struct SszNone {
+    value: Option<()>,
```

Comment:
What is a an `SszNone { value: None }` vs an `SszNone { value: Some(()) }`? Probably deserves a comment.